packages/protocol/contracts/signal/ISignalService.sol

```diff
@@ -50,24 +69,39 @@ interface ISignalService {
     )
         external;
 
+    /// @notice Verifies if a particular signal has already been sent.
+    /// @param app The address that initiated the signal.
+    /// @param signal The signal (message) that was sent.
+    /// @return True if the signal has been sent, otherwise false.
+    function isSignalSent(address app, bytes32 signal) external view returns (bool);
+
     /// @notice Checks if a chain data has been relayed.
     /// uniquely from chainId, kind, and data.
     /// @param chainId The remote chainId.
+    /// @param blockId The chain data's corresponding blockId
     /// @param kind A value to mark the data type.
-    /// @param data The remote data.
+    /// @param chainData The remote data.
     /// @return True if the data has been relayed, otherwise false.
     function isChainDataRelayed(
         uint64 chainId,
+        uint64 blockId,
         bytes32 kind,
-        bytes32 data
+        bytes32 chainData
     )
         external
         view
         returns (bool);
 
-    /// @notice Verifies if a particular signal has already been sent.
-    /// @param app The address that initiated the signal.
-    /// @param signal The signal (message) that was sent.
-    /// @return True if the signal has been sent, otherwise false.
-    function isSignalSent(address app, bytes32 signal) external view returns (bool);
+    /// @notice Returns the latest relayed chain data.
+    /// @param blockId The chain data's corresponding blockId
+    /// @param kind A value to mark the data type.
+    /// @return blockId The block id.
+    /// @return chainData The relayed chain data.
+    function getLatestChainData(
+        uint64 chainId,
+        bytes32 kind
```

Comment:
Without being able to specify the `blockId` it will be trickier to use this function onchain because transactions may quickly get outdated because different data will be queried each time the top blockId changes.

Should be okay as is if onchain querying of data from other smart contracts is not an expected usecase of this function.